x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/indicator_match/threat_mapping/create_threat_signals.ts

```diff
@@ -170,14 +174,56 @@ export const createThreatSignals = async ({
     let list = await getDocumentList({ searchAfter: undefined });
     let documentCount = totalDocumentCount;
 
+    // this is re-assigned depending on max clause count errors
+    let chunkPage = itemsPerSearch;
+
     while (list.hits.hits.length !== 0) {
       verifyExecutionCanProceed();
-      const chunks = chunk(itemsPerSearch, list.hits.hits);
+      const chunks = chunk(chunkPage, list.hits.hits);
       ruleExecutionLogger.debug(`${chunks.length} concurrent indicator searches are starting.`);
       const concurrentSearchesPerformed =
         chunks.map<Promise<SearchAfterAndBulkCreateReturnType>>(createSignal);
       const searchesPerformed = await Promise.all(concurrentSearchesPerformed);
-      results = combineConcurrentResults(results, searchesPerformed);
+
+      const maxClauseCountValue = getMaxClauseCountErrorValue(
+        searchesPerformed,
+        threatFieldsLength,
+        eventsTelemetry
+      );
+
+      if (maxClauseCountValue > Number.NEGATIVE_INFINITY) {
```

Comment:
I haven't run into this in my own testing surprisingly but yes it should be easy to add a bit more context to the error messages. Do you mind sharing the testing steps you took to get this? Haven't seen this yet but I suppose I could see how it would happen.

Reply:
I just followed steps you provided in the ticket's description earlier for kbn-alert-load tool

```
./kbn-alert-load.js run --type='im' -e http://elastic:changeme@localhost:9200 -k http://elastic:changeme@localhost:5601/kbn im-test    
```

config changes in `imSuites.js`

```

  eventCount: 10000,
  ruleCount: 2,
```